### PR TITLE
added workchunk to step execution details

### LINKED ISSUE #5919: Add workchunk to the jobexecution details
Currently only the workchunk id is passed forward.

But extra metadata exist on the work chunk (poll times, for instance) that might be of use.

### PR DESCRIPTION
**Pull request by TipzCM**
closes #5919

### CODE REVIEW

**Review by longma1 — APPROVED** [review]

File: hapi-fhir-storage-batch2/src/main/java/ca/uhn/fhir/batch2/api/StepExecutionDetails.java [hunk 1]
```diff
@@ -30,19 +32,19 @@ public class StepExecutionDetails<PT extends IModelJson, IT extends IModelJson>
 	private final PT myParameters;
 	private final IT myData;
 	private final IJobInstance myInstance;
-	private final String myChunkId;
+	private final WorkChunk myChunk;
 
 	public StepExecutionDetails(
 			@Nonnull PT theParameters,
```
[the remark this answers is not in the document]
Comment (TipzCM) [reply hapi-fhir-storage-batch2/src/main/java/ca/uhn/fhir/batch2/api/StepExecutionDetails.java:54]: added

File: hapi-fhir-storage-batch2-jobs/src/test/java/ca/uhn/fhir/batch2/jobs/imprt/FetchFilesStepTest.java [hunk 2]
```diff
@@ -59,7 +60,7 @@ public void testFetchWithBasicAuth(String theHeaderContentType) {
 		BulkImportJobParameters parameters = new BulkImportJobParameters()
 			.addNdJsonUrl(myHttpServletExtension.getBaseUrl() + "/download?index=" + index)
 			.setHttpBasicCredentials("admin:password");
-		StepExecutionDetails<BulkImportJobParameters, VoidModel> details = new StepExecutionDetails<>(parameters, null, ourTestInstance, CHUNK_ID);
+		StepExecutionDetails<BulkImportJobParameters, VoidModel> details = new StepExecutionDetails<>(parameters, null, ourTestInstance, new WorkChunk().setId(CHUNK_ID));
```
Comment (fil512) [inline hapi-fhir-storage-batch2-jobs/src/test/java/ca/uhn/fhir/batch2/jobs/imprt/FetchFilesStepTest.java:63]: Does this change the size of kafka messages or db records?

**Review by fil512 — CHANGES_REQUESTED** [review]
I need to understand performance implications before approving

File: hapi-fhir-storage-batch2/src/main/java/ca/uhn/fhir/batch2/util/Batch2Utils.java [hunk 3]
```diff
@@ -35,4 +42,24 @@ public class Batch2Constants {
...
+
+	/**
+	 * Create and returns a step execution details for a reduction job
+	 */
+	public static <P1 extends IModelJson, I1 extends IModelJson>
+			StepExecutionDetails<P1, I1> createReductionStepDetails(
+					P1 theP, I1 theIntermediateParams, JobInstance theInstance) {
+		WorkChunk reductionChunk = new WorkChunk().setId(REDUCTION_STEP_CHUNK_ID_PLACEHOLDER);
+
+		return new StepExecutionDetails<>(theP, theIntermediateParams, theInstance, reductionChunk);
```
[the remark this answers is not in the document]
Comment (TipzCM) [reply hapi-fhir-storage-batch2/src/main/java/ca/uhn/fhir/batch2/util/Batch2Utils.java:54]: Updated and moved to StepExecutionDetails

File: hapi-fhir-storage-batch2/src/main/java/ca/uhn/fhir/batch2/util/Batch2Utils.java [hunk 4]
```diff
@@ -35,4 +42,24 @@ public class Batch2Constants {
...
+	}
+
+	/**
+	 * Returns true if the workchunk is a reduction workchunk; false otherwise
+	 * @param theChunk the reduction work chunk.
+	 * @return true if reduction workchunk, false otherwise.
+	 */
+	public static boolean isReductionWorkChunk(@Nonnull WorkChunk theChunk) {
+		return theChunk.getId() != null && theChunk.getId().equals(REDUCTION_STEP_CHUNK_ID_PLACEHOLDER);
```
[the remark this answers is not in the document]
Comment (TipzCM) [reply hapi-fhir-storage-batch2/src/main/java/ca/uhn/fhir/batch2/util/Batch2Utils.java:63]: moved to workchunk

File: hapi-fhir-storage-batch2-jobs/src/test/java/ca/uhn/fhir/batch2/jobs/imprt/FetchFilesStepTest.java (hunk 2 above)
Comment (TipzCM) [reply hapi-fhir-storage-batch2-jobs/src/test/java/ca/uhn/fhir/batch2/jobs/imprt/FetchFilesStepTest.java:63]: Neither.

StepExecutionDetails are created during work step processing.

Nothing new is stored or sent.

**Review by fil512 — APPROVED** [review]

**Pull request merged**